src/webServer/webServer.ts

```diff
@@ -0,0 +1,193 @@
+        }
+
+        endGroup(): void {
+            this.currentGroupCount = Math.max(0, this.currentGroupCount - 1);
+        }
+
+        msg(s: string, type: Msg = Msg.Err): void {
+            s = `${type} ${this.seq.toString()} [${nowString()}] ${s}`;
+
+            switch (type) {
+                case Msg.Info:
+                    this.write("info", s);
+                    break;
+
+                case Msg.Perf:
+                    this.write("perf", s);
+                    break;
+
+                case Msg.Err:
+                default:
+                    this.write("error", s);
+                    break;
+            }
+
+            if (this.currentGroupCount === 0) {
+                this.seq++;
+            }
+        }
+
+        getLogFileName(): string | undefined {
+            return undefined;
+        }
+
+        private write(level: MessageLogLevel, body: string) {
+            this.host.writeMessage(<LoggingMessage>{
+                type: "log",
+                level,
+                body,
+            });
+        }
+    }
+
+    export function createWebSystem(host: WebHost, args: string[], getExecutingFilePath: () => string): ServerHost {
+        const returnEmptyString = () => "";
+        const getExecutingDirectoryPath = memoize(() => memoize(() => ensureTrailingDirectorySeparator(getDirectoryPath(getExecutingFilePath()))));
+        // Later we could map ^memfs:/ to do something special if we want to enable more functionality like module resolution or something like that
+        const getWebPath = (path: string) => startsWith(path, directorySeparator) ? path.replace(directorySeparator, getExecutingDirectoryPath()) : undefined;
+        return {
+            args,
+            newLine: "\r\n", // This can be configured by clients
+            useCaseSensitiveFileNames: false, // Use false as the default on web since that is the safest option
+            readFile: path => {
+                const webPath = getWebPath(path);
+                return webPath && host.readFile(webPath);
+            },
+
+            write: host.writeMessage.bind(host),
+            watchFile: returnNoopFileWatcher,
+            watchDirectory: returnNoopFileWatcher,
+
+            getExecutingFilePath: () => directorySeparator,
+            getCurrentDirectory: returnEmptyString, // For inferred project root if projectRoot path is not set, normalizing the paths
+
+            /* eslint-disable no-restricted-globals */
+            setTimeout,
+            clearTimeout,
+            setImmediate: x => setTimeout(x, 0),
+            clearImmediate: clearTimeout,
+            /* eslint-enable no-restricted-globals */
+
+            require: () => ({ module: undefined, error: new Error("Not implemented") }),
+            exit: notImplemented,
+
+            // Debugging related
+            getEnvironmentVariable: returnEmptyString, // TODO:: Used to enable debugging info
+            // tryEnableSourceMapsForHost?(): void;

```

Comment:
Consider cleaning up the comments.

Reply:
These comments are for future system methods categorized so helpful to have them here sorted and will like to retain them.